src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/LoopDetector.kt

```diff
@@ -222,6 +249,68 @@ internal class LoopDetector(
             Decision.Idle
     }
 
+    /**
+     * Called before regular method calls.
+     */
+    fun beforeRegularMethodCall(codeLocation: Int, params: Array<Any?>) {
+        replayModeLoopDetectorHelper?.let {
+            it.onNextExecution()
+            return
+        }
+        currentThreadCodeLocationsHistory += codeLocation
+        passParameters(params)
+        val lastInterleavingHistoryNode = currentInterleavingHistory.last()
+        if (lastInterleavingHistoryNode.cycleOccurred) {
+            return /* If we already ran into cycle and haven't switched than no need to track executions */
+        }
+        lastInterleavingHistoryNode.addExecution(codeLocation)
+        loopTrackingCursor.onNextExecutionPoint()
+    }
+
+    /**
+     * Called after any method calls.
+     */
+    fun afterRegularMethodCall() {
+        val afterMethodCallLocation = 0
```

Comment:
Because we want to track the fact of the method exit,
 but the sequence of the beforeMethodCall code locations values determines what exactly method we exit.
That's why we can use just 0 as a label of a method exit.

I added it as a comment.